### PR TITLE
templating for ReflectionClass

### PR DESCRIPTION
**Pull request by SignpostMarv**
I'm not 100% sure on this one, since it's mostly a case of being able to typehint `ReflectionClass<T>`- not entirely sure exactly how to spec out a test for it other than https://getpsalm.org/r/e73831ec8b

### CODE REVIEW

File: src/Psalm/Internal/Stubs/CoreGenericClasses.php [hunk 1]
```diff
@@ -1151,3 +1151,21 @@ public function offsetGet ($object) {}
...
+* @template T as object
+*
+* @property-read class-string<T> $name
+*/
+class ReflectionClass implements Reflector {
+
+    /**
+    * @var class-string<T>
+    */
+    public string $name;
```
Comment (weirdan) [inline src/Psalm/Internal/Stubs/CoreGenericClasses.php]: I don't think Psalm supports typed properties as of yet.

Comment (SignpostMarv) [reply src/Psalm/Internal/Stubs/CoreGenericClasses.php]: see https://mobile.twitter.com/psalmphp/status/1083856311689928705

Comment (weirdan) [reply src/Psalm/Internal/Stubs/CoreGenericClasses.php]: Ah, great, somehow missed that.

**Comment by SignpostMarv** [comment]
added instantation methods in c2c214d

**Comment by muglug** [comment]
Thanks!

**Pull request merged**